eng/Versions.props

```diff
@@ -78,6 +79,7 @@
   <PropertyGroup Label="Manual">
     <!-- Packages from dotnet/runtime -->
     <MicrosoftBclAsyncInterfacesLTSVersion>8.0.0</MicrosoftBclAsyncInterfacesLTSVersion>
+    <MicrosoftBclMemoryLTSVersion>9.0.1</MicrosoftBclMemoryLTSVersion>
```

Comment:
🤔 I think this should be 8.0.0, otherwise there's a risk of pulling non-LTS dependencies.

Reply:
Ah good point - I will switch it to 8.*

Reply:
@RussKie looks like `Microsoft.Bcl.Memory` is new. I only see 9.* versions here https://www.nuget.org/packages/Microsoft.Bcl.Memory#versions-body-tab. Any recommendations on how to fix this up?

Reply:
In this case I don't think we can use it in the LTS releases. You may need to copy the required functionality and secure it with `#if !NET9_0_OR_GREATER` pragmas.
There are examples under /src/Shared. If the code is needed by several projects, then it should go under Shared, otherwise it'd live in the project.

Reply:
On closer look looks like we are no longer using `Microsoft.Bcl.Memory`. I've gone ahead and removed it.